docs/docs/developers/versions-updating.md

```diff
@@ -0,0 +1,130 @@
+When developing and referring to example .nr files/snippets, it is helpful to verify the versions of different components (below), and if required keep them in lock-step by [updating](#updating).
+
+### Checking tool versions
+To check your version of Aztec tools, you can use `aztec-cli -V`
+::note
+The `aztec-nargo` versions follow `nargo` versions, which is different to the Aztec tool versions.
+::note
+The latest version of the Aztec tooling is currently `#include_aztec_version` , updating roughly every week.
+
+### Dependency versions
+Dependency versions in a contract's `Nargo.toml` file correspond to the `aztec-packages` repository tag `aztec-packages` (filter tags by `aztec`...)
+
+If you get an error like: `Cannot read file ~/nargo/github.com/AztecProtocol/aztec-packages/...`
+Check the `git=` github url, tag, and directory.
+:::note
+The folder structure changed at **0.24.0** from `yarn-project/aztec-nr` to `noir-projects/aztec-nr`.
+:::note
+
+That is, BEFORE `aztec-packages-v0.24.0`:
+`aztec = { git="https://github.com/AztecProtocol/aztec-packages/", tag="aztec-packages-v0.23.0", directory="yarn-project/aztec-nr/aztec" }`
+At/after `aztec-packages-v0.24.0`:
+`aztec = { git="https://github.com/AztecProtocol/aztec-packages/", tag="aztec-packages-v0.24.0", directory="noir-projects/aztec-nr/aztec" }`
+
+### Example contract versions
+Example contracts serve as an especially helpful reference between versions of the aztec-nr framework.
+
+Code referenced in the documentation is sourced from a directory in the `aztec-packages` repository.
+As in the previous section, the location of the noir contracts moved at version `0.24.0`, from `yarn-project/noir-contracts` before, to `noir-projects/noir-contracts`.
+
+:::tip
+Notice the difference between the sample Counter contract from `0.23.0` to `0.24.0` shows the `note_type_id` was added.
+`diff ~/nargo/github.com/AztecProtocol/aztec-packages-v0.23.0/yarn-project/noir-contracts/contracts/counter_contract/src/main.nr ~/nargo/github.com/AztecProtocol/aztec-packages-v0.24.0/noir-projects/noir-contracts/contracts/counter_contract/src/main.nr `
+```
+57a58
+>         note_type_id: Field,
+```
+:::tip
+
+### Language server version (nargo)
+The [Noir LSP](https://docs.aztec.network/developers/contracts/main#install-noir-lsp-recommended) uses your local version of `aztec-nargo`, and thus also `aztec-nargo compile`.
+The path of the former (once installed) can be seen by hovering over "Nargo" in the bottom status bar, and the latter via `which aztec-nargo`.
+
+This can present confusion when opening older noir contracts (and dependencies), such as:
+- Logs filled with errors from the dependencies
+- Or the LSP fails (re-runs automatically then stops)
+The second point requires a restart of the extension.
+
+:::tip
+When using the LSP in VSCode, reference older versions of Noir code in a separate window or a browser.
```

Comment:
How does a separate window help? Doesn't the LSP always use locally installed version of nargo?

Reply:
If you're using the correct version of aztec-nargo with the correct version of your code, all is well. If you happen to open an older versioned contract, and unlucky to hit some error with aztec-nargo and that code version, then vscode will try rerun the extension. This happens 5 times before the extension decides to stop trying.
At that point, even if you close the older example, you'll have to restart Noir LSP extension in the window.

To prevent this scenario the suggestion is to keep your primary window pure, and reference the old code elsewhere. If in another vscode window (not tab), the main window's extension is fine.

Happy to drop this since the "tip" tag is overly prominent for this minor/low-probability devexp optimisation.